app.js

```diff
@@ -66,6 +67,17 @@ if (app.get('env') === 'production') {
 app.use(session(session_opts));
 
 //==============================================================================
+// CSRF MIDDLEWARE
+//==============================================================================
+
+app.use(cookieParser());
+
+app.use((err, req, res, next) => {
```

Comment:
We are applying this on every request, shouldn't we limit it for the things we really need?